File: OpenCredentialPublisher.Data/ViewModels/nG/ClrSimplified/ClrVM.cs
using OpenCredentialPublisher.Data.Models;
using System;
using System.Collections.Generic;

namespace OpenCredentialPublisher.Data.ViewModels.nG.ClrSimplified
{
    public class ClrVM
    {
        public int Id { get; set; }
        public int PackageId { get; set; }
        public string Name { get; set; }
        public string PublisherName { get; set; }
        public bool IsRevoked { get; set; }
        public string Identifier { get; set; }
        public bool IsSelected { get; set; } = false;
        public string AuthorizationId { get; set; }
        public DateTime IssuedOn { get; set; }
        public DateTime RefreshedAt { get; set; }
        public int AssertionsCount { get; set; }
        public bool IsCollapsed { get; set; } = true;
        public List<string> AchievementIds { get; set; }
        public ProfileVM Learner { get; set; }
        public ProfileVM Publisher { get; set; }

        public static ClrVM FromModel(ClrModel clr, List<string> achievementIds = null)
        {
            var authId = clr.Authorization == null ? null : clr.Authorization.Id;
            return new ClrVM
            {
                Id = clr.ClrId,
                IsRevoked = clr.IsRevoked,
                IssuedOn = clr.IssuedOn,
                Name = clr.Name,
                PackageId = clr.CredentialPackageId,
                PublisherName = clr.PublisherName,
                AuthorizationId = authId,
                RefreshedAt = clr.RefreshedAt,
                AssertionsCount = clr.AssertionsCount,
                Identifier = clr.Id,
                AchievementIds = achievementIds ?? new List<string>(),
                Learner = ProfileVM.FromModel(clr.Learner),
                Publisher = ProfileVM.FromModel(clr.Publisher)
            };
        }
    }
}

File: OpenCredentialPublisher.Wallet/src/ClientApp/src/app/shared/models/clrVM.ts
import { ProfileVM } from "./clrSimplified/profileVM";
import { StatusEnum } from "./enums/statusEnum";

export class ClrVM {
  id = -1;
  packageId = -1;
  name = '';
  authorizationId = '';
  publisherName = '';
  isRevoked = false;
  isSelected = false;
  issuedOn: Date;
  refreshedAt: Date;
  assertionsCount: number;
  isCollapsed = false;
  identifier = '';
  achievementIds: string[];
  enableSmartResume: boolean = false;
  hasSmartResume: boolean = false;
  smartResumeUrl: string = '';
  smartResumeStatus: StatusEnum | null;
  smartResumeMessage: string = '';
  learner: ProfileVM;
  publisher: ProfileVM;
  constructor(){
    this.id = -1;
    this.packageId = -1;
    this.name = '';
    this.authorizationId = '';
    this.publisherName = '';
    this.isRevoked = false;
    this.isSelected = false;
    this.issuedOn = null;
    this.refreshedAt = null;
    this.assertionsCount = -1;
    this.isCollapsed = false;
    this.identifier = '';
    this.achievementIds = new Array<string>();
    this.learner = null;
    this.publisher = null;
    this.enableSmartResume = false;
    this.hasSmartResume = false;
    this.smartResumeStatus = null;
    this.smartResumeMessage = null;
    this.smartResumeUrl = '';
  }
}
